dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsImplCommon.java

```diff
@@ -266,33 +268,49 @@ public static Tool convertEntryToTool(Entry<?, ?> container, DockstoreWebservice
      * @param version tag for Dockstore tool
      * @param toolVersion toolVersion to return in TRS
      */
-    private static void processImageDataForToolVersion(io.dockstore.webservice.core.Tool castedContainer, Tag version,
+    public static void processImageDataForToolVersion(io.dockstore.webservice.core.Tool castedContainer, Tag version,
         ToolVersion toolVersion) {
-        ImageData data = new ImageData();
         List<Checksum> trsChecksums = new ArrayList<>();
-        if (version.getImages() != null) {
+        if (version.getImages() != null && !version.getImages().isEmpty()) {
             version.getImages().forEach(image -> {
+                ImageData data = new ImageData();
                 image.getChecksums().forEach(checksum -> {
                     Checksum trsChecksum = new Checksum();
                     trsChecksum.setType(DOCKER_IMAGE_SHA_TYPE_FOR_TRS);
                     trsChecksum.setChecksum(checksum.getChecksum());
                     trsChecksums.add(trsChecksum);
                 });
+                //TODO: for now, all container images are Docker based
+                data.setImageType(ImageType.DOCKER);
+                data.setSize(image.getSize());
+                data.setUpdated(new Date().toString());
+                data.setImageName(constructName(Arrays.asList(castedContainer.getRegistry(), castedContainer.getNamespace(), castedContainer.getName())));
+                data.setRegistryHost(castedContainer.getRegistry());
+                data.setChecksum(trsChecksums);
+                toolVersion.getImages().add(data);
             });
-            data.setChecksum(trsChecksums);
         } else {
-            //tools without images?
-            data.setChecksum(MoreObjects.firstNonNull(data.getChecksum(), Lists.newArrayList()));
+            toolVersion.getImages().add(createDummyImageData(castedContainer));
         }
+    }
+
+    /**
+     * This is a workaround used when the version.getImages() has nothing in it
+     * @param castedContainer
+     * @return
+     */
+    private static ImageData createDummyImageData(io.dockstore.webservice.core.Tool castedContainer) {
+        ImageData data = new ImageData();
         //TODO: for now, all container images are Docker based
         data.setImageType(ImageType.DOCKER);
-        //TODO: hook up proper size
-        data.setSize(0);
-        //TODO: hook up proper date
+        // Not a proper size because we don't have that information stored in version.getImages()
+        data.setSize(0L);
+        // Not a proper date because we don't have that information stored in version.getImages()
         data.setUpdated(new Date().toString());
```

Comment:
Ah, I see the date was already like this. 

1. I still think it should be an ISO-8601 format.
2. Seems like this code and the above code could be refactored/combined.

Reply:
1. I've changed it to a variation of the many ISO-8601 even though it doesn't matter (dummy data)
2. That gets messy and is not worth it. Constructing a partial object where checksum, size and date are different